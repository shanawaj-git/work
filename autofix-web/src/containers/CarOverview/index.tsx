import PageHeader from "@/components/PageHeader/PageHeader";
import Section from "@/components/Section";
import { Article, Button, utils } from "@albathanext/design-system";
import React, { ReactElement, useCallback, useEffect, useState } from "react";
import { FormattedMessage, useIntl } from "react-intl";
import BackArrow from "@/assets/arrow-back-icon.svg";
import EditIcon from "@/assets/edit.svg";
import { useRouter } from "next/router";
import { LabelType } from "@/services/common";
import { CarType } from "@/Types/Car";
import { getCarDetails } from "@/services/carOverview";
import get from "lodash.get";
import Loading from "@/components/Loading/Loading";
import { localStorage } from "@/utils/storage/LocalStorage";
import { CAR_SELECTION, RECOMMENDED_MAINTENANCE } from "@/app/config";
export const pageName = "car.overview";
export const USER_SELECTED_CAR_DETAILS = "user_selected_car_details";
interface CarDetailType extends CarType {
  mileage: string;
}

export interface CarOverviewSelectionPropsType {
  en?: { labels: LabelType[] };
  ar?: { labels: LabelType[] };
}

const CarOverview = () => {
  const router = useRouter();
  const [selectedCar, setSelectedCar] = useState<CarDetailType>();
  const [loading, setLoading] = useState<boolean>(false);
  const intl = useIntl();

  const getSelectedCarDetails = useCallback(async () => {
    setLoading(true);
    const { modelId, yearId, engineId, mileage } = router.query;

    const car: CarType = await getCarDetails({
      modelId: modelId as string,
      yearId: yearId as string,
      engineId: engineId as string,
    });

    const carDetails = {
      ...car,
      mileage: mileage as string,
      ...(!engineId && {
        engineType: {
          id: engineId as string,
          name: intl.formatMessage({ id: "car.overview.engine.none" }),
        },
      }),
    };
    setSelectedCar(carDetails);

    setLoading(false);
  }, [router.query]);

  useEffect(() => {
    if (router.isReady) {
      getSelectedCarDetails();
    }
  }, [router.isReady, getSelectedCarDetails]);

  const handleOnClickBack = () => {
    router.back();
  };

  const onEditClickHandler = () => {
    router.push(CAR_SELECTION);
  };

  const onCarConfirmationHandler = () => {
    const { setItem } = localStorage();
    setItem(USER_SELECTED_CAR_DETAILS, JSON.stringify(selectedCar));
    router.push(RECOMMENDED_MAINTENANCE);
  };

  const getNameOfTheCar = (
    make: string,
    model: string,
    year: string | number
  ): string => {
    return `${make} ${model} ${year}`;
  };

  if (loading) {
    return <Loading />;
  }

  return (
    <Section
      className={`flex flex-col  h-screen`}
      style={{
        backgroundColor: utils.colorSelector("C500"),
      }}
      data-testid={`${pageName}`}
    >
      <PageHeader>
        <Button
          data-testid={`${pageName}-back-button`}
          className={`mb-5 min-w-[40px] min-h-[40px] w-[40px] h-[40px]`}
          style={{
            borderColor: "rgba(220, 220, 220, 0.2)",
            backgroundColor: utils.colorSelector("P700"),
          }}
          onClick={handleOnClickBack}
        >
          <img src={BackArrow} alt="BackArrow" />
        </Button>
        <Article
          title={<FormattedMessage id={`${pageName}.header.title`} />}
          titleProps={{
            color: utils.colorSelector("P200"),
            className: "text-xl whitespace-pre-line !mt-4",
          }}
          description={<FormattedMessage id={`${pageName}.header.subTitle`} />}
          descriptionProps={{
            className: "text-white/[0.8] text-sm mt-[12px]",
          }}
        />
      </PageHeader>

      <div
        className="flex flex-col items-center justify-center"
        style={{ backgroundColor: utils.colorSelector("C500") }}
      >
        <div className="mx-[28px] mt-[52px]">
          <img
            data-testid="car-image"
            src={get(selectedCar, "image")}
            width="320px"
            height="166px"
            alt={`vehicle image`}
          />
        </div>

        <div className="flex mt-[54px] px-[20px] justify-between w-full">
          <div
            data-testid="car-make-model-year"
            className="w-[245px] text-[24px] font-semibold"
          >
            {getNameOfTheCar(
              get(selectedCar, "maker.name", ""),
              get(selectedCar, "model.name", ""),
              get(selectedCar, "year.value", "")
            )}
          </div>
          <div className="w-[41px] h-[27px] pt-1">
            <img
              data-testid="car-make-logo"
              src={get(selectedCar, "maker.logo")}
              alt={`${get(selectedCar, "maker.name")} logo`}
            />
          </div>
        </div>

        <div className="flex px-[20px] w-full my-[20px]">
          <div className="flex flex-col w-1/2">
            <CarOverviewItemComponent
              itemName="car-engine-type"
              label={<FormattedMessage id={`${pageName}.details.engine`} />}
              value={`${get(selectedCar, "engineType.name", "")}`}
            />
          </div>
          <div className="flex flex-col w-1/2 border-l-2 pl-[20px]">
            <CarOverviewItemComponent
              itemName="car-mileage"
              label={<FormattedMessage id={`${pageName}.details.mileage`} />}
              value={`${get(selectedCar, "mileage", "0")} km`}
            />
          </div>
        </div>

        <div className="flex px-[20px] w-full my-[47px]">
          <div className="w-1/2 text-[14px] font-semibold pt-[15px] cursor-pointer">
            <div
              data-testid="edit-car-details-link"
              className="inline-block text-[#2E4341]"
              onClick={onEditClickHandler}
            >
              <FormattedMessage id={`${pageName}.details.edit`} />
              <img
                className="inline-block px-[4px] pb-[2px]"
                src={EditIcon}
                alt="edit icon"
              />
            </div>
          </div>
          <div className="w-1/2">
            <Button
              data-testid={`${pageName}-button-confirm`}
              variant="contained"
              fullWidth
              size="medium"
              className="!text-[13px] !px-[17.25px] !normal-case whitespace-nowrap font-semibold"
              onClick={onCarConfirmationHandler}
            >
              <FormattedMessage id={`${pageName}.button.confirm`} />
            </Button>
          </div>
        </div>
      </div>
    </Section>
  );
};

interface CarOverviewItemComponentType {
  itemName: string;
  label: string | ReactElement;
  value: string;
}

const CarOverviewItemComponent = ({
  itemName,
  label,
  value,
}: CarOverviewItemComponentType) => {
  return (
    <>
      <div
        data-testid={`${itemName}-label`}
        className="text-[16px] text-black/60 font-light"
      >
        {label}
      </div>
      <div data-testid={itemName} className="text-[20px] font-normal">
        {value}
      </div>
    </>
  );
};

export default CarOverview;
