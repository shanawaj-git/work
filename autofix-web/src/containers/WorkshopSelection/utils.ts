import {
  BoundPayloadType,
  GetDirections,
  LatLngType,
  MapObjType,
} from "./interfaces";

export const markerIcon = (icon: string) => ({
  scaledSize: {
    height: 24,
    width: 24,
  },
  url: icon,
});

export const getBoundLocation = ({
  map,
  onBoundChange,
}: {
  map: MapObjType;
  onBoundChange: (e: BoundPayloadType) => void;
}) => {
  const bounds = map.getBounds();
  const southWest = getSouthWestCoordinates(bounds);
  const northEast = getNorthEastCoordinates(bounds);
  const result = buildBoundPayload(northEast, southWest);
  onBoundChange(result);
};

const getSouthWestCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getSouthWest().lat(),
  lng: bounds.getSouthWest().lng(),
});

const getNorthEastCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getNorthEast().lat(),
  lng: bounds.getNorthEast().lng(),
});

const buildBoundPayload = (
  northEast: LatLngType,
  southWest: LatLngType
): BoundPayloadType => ({
  locationBoxBounds: {
    bottomLeft: {
      type: "Point",
      coordinates: [southWest.lng, southWest.lat],
    },
    topRight: { type: "Point", coordinates: [northEast.lng, northEast.lat] },
  },
});
