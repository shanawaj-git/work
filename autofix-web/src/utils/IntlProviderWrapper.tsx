import React, { useMemo } from "react";
import { getTranslation, TranslationsType } from "../translations";
import { IntlProvider } from "react-intl";
import { flatData, I18NLocale } from ".";

export const withWrapper = (component: JSX.Element, pageProps = { en: {} }) => (
  <IntlProviderWrapper locale="en" pageProps={pageProps}>
    {component}
  </IntlProviderWrapper>
);

export const IntlProviderWrapper = ({
  children,
  pageProps,
  locale,
}: {
  children: JSX.Element;
  pageProps: TranslationsType;
  locale: I18NLocale;
}) => {
  const messages = useMemo(() => {
    return {
      ...getTranslation(locale),
      ...flatData(pageProps[locale] || {}),
    };
  }, [locale, pageProps]);

  return (
    <IntlProvider locale={`${locale}`} messages={messages}>
      {children}
    </IntlProvider>
  );
};
