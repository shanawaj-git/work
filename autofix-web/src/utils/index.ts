import { itemType } from "@/containers/RecommendedMaintenance";
import {
  AssetType,
  fetchCommonData,
  FooterType,
  getAssets,
  getAssetsData,
  getLabelData,
  getLabels,
  HomeActionType,
  LabelType,
  ServiceType,
} from "@/services/common";
import { TestimonialType } from "@/services/home/testimonial";

export const utils = "UTILS";
const GRAPHQL_CLIENT_READ_TIMEOUT = 5000;

interface getCommonPageContentParams {
  locale: I18NLocale;
  pageName: string;
}

export interface HomeContentType {
  labels?: LabelType[];
  assets?: AssetType[];
  services?: ServiceType[];
  actions?: HomeActionType[];
  testimonials?: TestimonialType[];
  footer?: FooterType;
}

export type I18NLocale = "en" | "ar";

/**
 * Will Return a Data object with Labels and Assets Regarding the page from Strapi
 * @param  locale , pageName
 * @returns Object with Labels and Assets
 */

export const getCommonPageContent = async ({
  locale,
  pageName,
}: getCommonPageContentParams): Promise<HomeContentType> => {
  let data: HomeContentType = {};
  return new Promise(async (resolve, reject) => {
    try {
      const commonData = await (
        await fetchCommonData(locale as I18NLocale, pageName)
      ).data;
      const assets = getAssets(commonData);
      const assetsData = getAssetsData(assets);
      const labels = getLabelData(getLabels(commonData));
      data = { assets: assetsData, labels };
      resolve(data);
    } catch (error) {
      resolve(data);
    }
  });
};

interface rejectEarlyType<T> {
  fn: () => Promise<T>;
  def: T;
  timeout?: number;
}

export const rejectEarly = <T>({
  fn,
  def,
  timeout = GRAPHQL_CLIENT_READ_TIMEOUT,
}: rejectEarlyType<T>): Promise<T> => {
  return new Promise(async (resolve, _reject) => {
    setTimeout(() => {
      resolve(def);
    }, timeout);
    try {
      const out: T = await fn();
      resolve(out);
    } catch (error) {
      resolve(def);
    }
  });
};

export const validateEmail = (email: string) =>
  /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email);

export const flatData = (obj: { [key: string]: {} }) =>
  Object.values(obj).reduce((acc: {}, curr: {}) => {
    return { ...acc, ...curr };
  }, {});

export const buildImageUrl = (path: string): string =>
  process.env.NEXT_PUBLIC_STRAPI_SERVICE?.concat(path) || "";

export const groupBy = <T>(
  array: T[],
  predicate: (value: T, index: number, array: T[]) => string
) =>
  array.reduce((acc, value, index, array) => {
    (acc[predicate(value, index, array)] ||= []).push(value);
    return acc;
  }, {} as { [key: string]: T[] });

export const includesCaseInsensitive = (
  string: string,
  substring: string
): Boolean => {
  return string.toLowerCase().includes(substring.toLowerCase());
};
export const getTotalForMax = (data: itemType[]) => {
  return data.reduce((acc: number, curr: { cost: string }) => {
    let value = acc;
    value += +parseFloat(
      (curr.cost.match(/[0-9]+/gm) || ["0"]).splice(-1)[0]
    ).toFixed(2);
    return value;
  }, 0);
};

export const getCurrency = (data: itemType[]) => {
  return (data[0].cost.match(/[A-Z]+/gm) || ["USD"]).splice(-1)[0];
};

export const getCurrentLocation = () => {
  return new Promise((resolve: (e: GeolocationCoordinates) => void, reject) => {
    if (!navigator) throw new Error("Navigator doesn't exist");
    if (!navigator.geolocation) throw new Error("Geolocation doesn't exist");
    navigator.geolocation.getCurrentPosition(
      (position: GeolocationPosition) => {
        return resolve(position.coords);
      },
      reject,
      {
        maximumAge: 5000,
        enableHighAccuracy: true,
      }
    );
  });
};

interface getCurrentLiveLocationType {
  callBack: (e: GeolocationPosition) => void;
  onError: (e: GeolocationPositionError | Error) => void;
}

export const getCurrentLiveLocation = ({
  callBack,
  onError,
}: getCurrentLiveLocationType) => {
  if (!navigator) throw new Error("Navigator doesn't exist ");
  if (!navigator.geolocation) throw new Error("Geolocation doesn't exist ");
  navigator.geolocation.watchPosition((position: GeolocationPosition) => {
    callBack(position);
  }, onError);
};
