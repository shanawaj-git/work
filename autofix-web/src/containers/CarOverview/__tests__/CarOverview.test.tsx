import React from "react";
import { act, fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CarOverview, { pageName, USER_SELECTED_CAR_DETAILS } from "..";
import { withWrapper } from "@/utils/IntlProviderWrapper";
import { useRouter } from "next/router";
import * as mockCarsQueryResponse from "./CarsQueryResponse.json";
import client from "@/apollo/index";

const asyncRenderComponent = async () => {
  let tree;
  await act(async () => {
    tree = render(withWrapper(<CarOverview />));
  });
  return tree;
};

const selectedCarDetails = {
  engineType: {
    id: "69",
    name: "1.6 VVT-i",
  },
  model: {
    id: "23",
    name: "Corolla",
  },
  maker: {
    id: "10",
    name: "Toyota",
    logo: "https://banner2.cleanpng.com/20171201/3ce/toyota-logo-picture-5a21ff4daacf07.7378398915121774856996.jpg",
  },
  year: {
    id: "49",
    value: 2022,
  },
  image:
    "https://www.drivearabia.com/carprices/media/catalog/5d0b8000581442020_Toyota_Corolla_front.jpg",
  mileage: "15000",
};

describe("<CarOverview /> ", () => {
  const router = useRouter();
  router.isReady = true;

  const carOverviewTests = () => {
    describe("Car Overview test cases", () => {
      test("it should mount", async () => {
        const screen = await asyncRenderComponent();
        const carOverview = screen.getByTestId(pageName);
        expect(carOverview).toBeInTheDocument();
      });

      test("it should mount and match to the snapshot", async () => {
        const tree = await asyncRenderComponent();
        expect(tree.container.innerHTML).toMatchSnapshot();
      });

      test("it should render car vehicle image with alternative text", async () => {
        const tree = await asyncRenderComponent();
        const element = tree.getByTestId("car-image");
        expect(element).toBeInTheDocument();
        expect(element.getAttribute("alt")).toBe("vehicle image");
      });

      test("it should render car name with make model year combination", async () => {
        const tree = await asyncRenderComponent();
        const element = tree.getByTestId("car-make-model-year");
        expect(element).toBeInTheDocument();
        expect(element.innerHTML).toBe("Toyota Corolla 2022");
      });

      test("it should render make logo", async () => {
        const tree = await asyncRenderComponent();
        const element = tree.getByTestId("car-make-logo");
        expect(element).toBeInTheDocument();
        expect(element.getAttribute("alt")).toBe("Toyota logo");
      });

      test("it should render car mileage", async () => {
        const tree = await asyncRenderComponent();
        const labelElement = tree.getByTestId("car-mileage-label");
        const valueElement = tree.getByTestId("car-mileage");
        expect(labelElement).toBeInTheDocument();
        expect(labelElement.innerHTML).toBe("Mileage");
        expect(valueElement).toBeInTheDocument();
        expect(valueElement.innerHTML).toBe("15000 km");
      });

      test("it should render edit car details link and it should be clickable", async () => {
        const tree = await asyncRenderComponent();
        const editElement = tree.getByTestId("edit-car-details-link");
        expect(editElement).toBeInTheDocument();
        expect(editElement.innerHTML).toContain("Edit car details");
        fireEvent(editElement, new MouseEvent("click", { bubbles: true }));
        expect(router.push).toHaveBeenCalled();
      });

      test("verify confirm buton is on dom and enabled ", async () => {
        const tree = await asyncRenderComponent();
        const buttonElement = tree.getByTestId(`${pageName}-button-confirm`);
        expect(buttonElement).toBeInTheDocument();
        expect(buttonElement.textContent).toContain("Confirm car details");
        expect(buttonElement).toBeEnabled();
      });

      test("when user clicks on confirm car details, it should redirect to recommended-maintenance ", async () => {
        const tree = await asyncRenderComponent();
        const buttonElement = tree.getByTestId(`${pageName}-button-confirm`);
        fireEvent(buttonElement, new MouseEvent("click", { bubbles: true }));
        expect(router.push).toBeCalledWith("/recommended-maintenance");
      });

      test("when user clicks back button, should redirect to previous route and it calls only once", async () => {
        const tree = await asyncRenderComponent();
        const button = tree.getByTestId(`${pageName}-back-button`);
        fireEvent(button, new MouseEvent("click", { bubbles: true }));
        expect(router.back).toBeCalledTimes(1);
      });
    });
  };

  const engineNotSpecifiedTest = () => {
    describe("Engine Type Not Specified", () => {
      test("it should render engine type", async () => {
        const tree = await asyncRenderComponent();
        const labelElement = tree.getByTestId("car-engine-type-label");
        const valueElement = tree.getByTestId("car-engine-type");
        expect(labelElement).toBeInTheDocument();
        expect(labelElement.innerHTML).toBe("Engine");
        expect(valueElement).toBeInTheDocument();
        expect(valueElement.innerHTML).toBe("Not Specified");
      });

      test("when user clicks on confirm car details button, it should save the data in localStorage ", async () => {
        const tree = await asyncRenderComponent();
        const buttonElement = tree.getByTestId(`${pageName}-button-confirm`);
        fireEvent(buttonElement, new MouseEvent("click", { bubbles: true }));
        const selectedCar = {
          ...selectedCarDetails,
          engineType: { name: "Not Specified" },
        };
        expect(window.localStorage.getItem(USER_SELECTED_CAR_DETAILS)).toBe(
          JSON.stringify(selectedCar)
        );
      });
    });
  };

  describe("when modelId, YearId and EngineId present in query params", () => {
    beforeEach(() => {
      client.query = jest.fn();
      client.query.mockResolvedValue(mockCarsQueryResponse);
      router.query = {
        mileage: "15000",
        yearId: "49",
        modelId: "23",
        engineId: "69",
      };
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    carOverviewTests();

    test("it should render engine type", async () => {
      const tree = await asyncRenderComponent();
      const labelElement = tree.getByTestId("car-engine-type-label");
      const valueElement = tree.getByTestId("car-engine-type");
      expect(labelElement).toBeInTheDocument();
      expect(labelElement.innerHTML).toBe("Engine");
      expect(valueElement).toBeInTheDocument();
      expect(valueElement.innerHTML).toBe("1.6 VVT-i");
    });

    test("when user clicks on confirm car details button, it should save the data in localStorage ", async () => {
      const tree = await asyncRenderComponent();
      const buttonElement = tree.getByTestId(`${pageName}-button-confirm`);
      fireEvent(buttonElement, new MouseEvent("click", { bubbles: true }));
      expect(window.localStorage.getItem(USER_SELECTED_CAR_DETAILS)).toBe(
        JSON.stringify(selectedCarDetails)
      );
    });
  });

  describe("when modelId, YearId and default EngineId present in query params", () => {
    beforeEach(() => {
      client.query = jest.fn();
      client.query.mockResolvedValue(mockCarsQueryResponse);
      router.query = {
        mileage: "15000",
        yearId: "49",
        modelId: "23",
      };
    });

    afterEach(() => {
      jest.clearAllMocks();
    });
    carOverviewTests();
    engineNotSpecifiedTest();
  });

  describe("when modelId, YearId combination is wrong and default EngineId present in query params", () => {
    beforeEach(() => {
      client.query = jest.fn();
      client.query.mockResolvedValue(mockCarsQueryResponse);
      router.query = {
        mileage: "15000",
        modelId: "23",
      };
    });

    afterEach(() => {
      jest.clearAllMocks();
    });
    carOverviewTests();
    engineNotSpecifiedTest();
  });
});
