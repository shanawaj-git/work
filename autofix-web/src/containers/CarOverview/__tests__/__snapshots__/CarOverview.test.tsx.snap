// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarOverview />  when modelId, YearId and EngineId present in query params Car Overview test cases it should mount and match to the snapshot 1`] = `"<div data-testid=\\"car.overview\\" class=\\"min-h-screen flex flex-col  h-screen\\"><div data-testid=\\"PageHeader\\" class=\\"min-h-screen p-6 \\" style=\\"min-height: fit-content;\\"><button class=\\"root StyledButton-sc-1apep4b-0 bXswYh mb-5 min-w-[40px] min-h-[40px] w-[40px] h-[40px] MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"car.overview-back-button\\" style=\\"border-color: rgba(220, 220, 220, 0.2);\\"><div style=\\"visibility: visible;\\"><img src=\\"arrow-back-icon.svg\\" alt=\\"BackArrow\\"></div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-xl whitespace-pre-line !mt-4 css-1uw3dvl-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Confirm your car model</span></p><p class=\\"MuiTypography-root MuiTypography-body1 text-white/[0.8] text-sm mt-[12px] css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-description\\"><span> This car will be saved to your profile.</span></p></div></div><div class=\\"flex flex-col items-center justify-center\\"><div class=\\"mx-[28px] mt-[52px]\\"><img data-testid=\\"car-image\\" src=\\"https://www.drivearabia.com/carprices/media/catalog/5d0b8000581442020_Toyota_Corolla_front.jpg\\" width=\\"320px\\" height=\\"166px\\" alt=\\"vehicle image\\"></div><div class=\\"flex mt-[54px] px-[20px] justify-between w-full\\"><div data-testid=\\"car-make-model-year\\" class=\\"w-[245px] text-[24px] font-semibold\\">Toyota Corolla 2022</div><div class=\\"w-[41px] h-[27px] pt-1\\"><img data-testid=\\"car-make-logo\\" src=\\"https://banner2.cleanpng.com/20171201/3ce/toyota-logo-picture-5a21ff4daacf07.7378398915121774856996.jpg\\" alt=\\"Toyota logo\\"></div></div><div class=\\"flex px-[20px] w-full my-[20px]\\"><div class=\\"flex flex-col w-1/2\\"><div data-testid=\\"car-engine-type-label\\" class=\\"text-[16px] text-black/60 font-light\\">Engine</div><div data-testid=\\"car-engine-type\\" class=\\"text-[20px] font-normal\\">1.6 VVT-i</div></div><div class=\\"flex flex-col w-1/2 border-l-2 pl-[20px]\\"><div data-testid=\\"car-mileage-label\\" class=\\"text-[16px] text-black/60 font-light\\">Mileage</div><div data-testid=\\"car-mileage\\" class=\\"text-[20px] font-normal\\">15000 km</div></div></div><div class=\\"flex px-[20px] w-full my-[47px]\\"><div class=\\"w-1/2 text-[14px] font-semibold pt-[15px] cursor-pointer\\"><div data-testid=\\"edit-car-details-link\\" class=\\"inline-block text-[#2E4341]\\">Edit car details<img class=\\"inline-block px-[4px] pb-[2px]\\" src=\\"edit.svg\\" alt=\\"edit icon\\"></div></div><div class=\\"w-1/2\\"><button class=\\"root StyledButton-sc-1apep4b-0 jVvup !text-[13px] !px-[17.25px] !normal-case whitespace-nowrap font-semibold MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"car.overview-button-confirm\\"><div style=\\"visibility: visible;\\">Confirm car details</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></div>"`;

exports[`<CarOverview />  when modelId, YearId and default EngineId present in query params Car Overview test cases it should mount and match to the snapshot 1`] = `"<div data-testid=\\"car.overview\\" class=\\"min-h-screen flex flex-col  h-screen\\"><div data-testid=\\"PageHeader\\" class=\\"min-h-screen p-6 \\" style=\\"min-height: fit-content;\\"><button class=\\"root StyledButton-sc-1apep4b-0 bXswYh mb-5 min-w-[40px] min-h-[40px] w-[40px] h-[40px] MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"car.overview-back-button\\" style=\\"border-color: rgba(220, 220, 220, 0.2);\\"><div style=\\"visibility: visible;\\"><img src=\\"arrow-back-icon.svg\\" alt=\\"BackArrow\\"></div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-xl whitespace-pre-line !mt-4 css-1uw3dvl-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Confirm your car model</span></p><p class=\\"MuiTypography-root MuiTypography-body1 text-white/[0.8] text-sm mt-[12px] css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-description\\"><span> This car will be saved to your profile.</span></p></div></div><div class=\\"flex flex-col items-center justify-center\\"><div class=\\"mx-[28px] mt-[52px]\\"><img data-testid=\\"car-image\\" src=\\"https://www.drivearabia.com/carprices/media/catalog/5d0b8000581442020_Toyota_Corolla_front.jpg\\" width=\\"320px\\" height=\\"166px\\" alt=\\"vehicle image\\"></div><div class=\\"flex mt-[54px] px-[20px] justify-between w-full\\"><div data-testid=\\"car-make-model-year\\" class=\\"w-[245px] text-[24px] font-semibold\\">Toyota Corolla 2022</div><div class=\\"w-[41px] h-[27px] pt-1\\"><img data-testid=\\"car-make-logo\\" src=\\"https://banner2.cleanpng.com/20171201/3ce/toyota-logo-picture-5a21ff4daacf07.7378398915121774856996.jpg\\" alt=\\"Toyota logo\\"></div></div><div class=\\"flex px-[20px] w-full my-[20px]\\"><div class=\\"flex flex-col w-1/2\\"><div data-testid=\\"car-engine-type-label\\" class=\\"text-[16px] text-black/60 font-light\\">Engine</div><div data-testid=\\"car-engine-type\\" class=\\"text-[20px] font-normal\\">Not Specified</div></div><div class=\\"flex flex-col w-1/2 border-l-2 pl-[20px]\\"><div data-testid=\\"car-mileage-label\\" class=\\"text-[16px] text-black/60 font-light\\">Mileage</div><div data-testid=\\"car-mileage\\" class=\\"text-[20px] font-normal\\">15000 km</div></div></div><div class=\\"flex px-[20px] w-full my-[47px]\\"><div class=\\"w-1/2 text-[14px] font-semibold pt-[15px] cursor-pointer\\"><div data-testid=\\"edit-car-details-link\\" class=\\"inline-block text-[#2E4341]\\">Edit car details<img class=\\"inline-block px-[4px] pb-[2px]\\" src=\\"edit.svg\\" alt=\\"edit icon\\"></div></div><div class=\\"w-1/2\\"><button class=\\"root StyledButton-sc-1apep4b-0 jVvup !text-[13px] !px-[17.25px] !normal-case whitespace-nowrap font-semibold MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"car.overview-button-confirm\\"><div style=\\"visibility: visible;\\">Confirm car details</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></div>"`;

exports[`<CarOverview />  when modelId, YearId combination is wrong and default EngineId present in query params Car Overview test cases it should mount and match to the snapshot 1`] = `"<div data-testid=\\"car.overview\\" class=\\"min-h-screen flex flex-col  h-screen\\"><div data-testid=\\"PageHeader\\" class=\\"min-h-screen p-6 \\" style=\\"min-height: fit-content;\\"><button class=\\"root StyledButton-sc-1apep4b-0 bXswYh mb-5 min-w-[40px] min-h-[40px] w-[40px] h-[40px] MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"car.overview-back-button\\" style=\\"border-color: rgba(220, 220, 220, 0.2);\\"><div style=\\"visibility: visible;\\"><img src=\\"arrow-back-icon.svg\\" alt=\\"BackArrow\\"></div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><div data-testid=\\"article-component\\"><p class=\\"MuiTypography-root MuiTypography-body1 text-xl whitespace-pre-line !mt-4 css-1uw3dvl-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Confirm your car model</span></p><p class=\\"MuiTypography-root MuiTypography-body1 text-white/[0.8] text-sm mt-[12px] css-1reifv3-MuiTypography-root\\" data-testid=\\"article-component-description\\"><span> This car will be saved to your profile.</span></p></div></div><div class=\\"flex flex-col items-center justify-center\\"><div class=\\"mx-[28px] mt-[52px]\\"><img data-testid=\\"car-image\\" src=\\"https://www.drivearabia.com/carprices/media/catalog/5d0b8000581442020_Toyota_Corolla_front.jpg\\" width=\\"320px\\" height=\\"166px\\" alt=\\"vehicle image\\"></div><div class=\\"flex mt-[54px] px-[20px] justify-between w-full\\"><div data-testid=\\"car-make-model-year\\" class=\\"w-[245px] text-[24px] font-semibold\\">Toyota Corolla 2022</div><div class=\\"w-[41px] h-[27px] pt-1\\"><img data-testid=\\"car-make-logo\\" src=\\"https://banner2.cleanpng.com/20171201/3ce/toyota-logo-picture-5a21ff4daacf07.7378398915121774856996.jpg\\" alt=\\"Toyota logo\\"></div></div><div class=\\"flex px-[20px] w-full my-[20px]\\"><div class=\\"flex flex-col w-1/2\\"><div data-testid=\\"car-engine-type-label\\" class=\\"text-[16px] text-black/60 font-light\\">Engine</div><div data-testid=\\"car-engine-type\\" class=\\"text-[20px] font-normal\\">Not Specified</div></div><div class=\\"flex flex-col w-1/2 border-l-2 pl-[20px]\\"><div data-testid=\\"car-mileage-label\\" class=\\"text-[16px] text-black/60 font-light\\">Mileage</div><div data-testid=\\"car-mileage\\" class=\\"text-[20px] font-normal\\">15000 km</div></div></div><div class=\\"flex px-[20px] w-full my-[47px]\\"><div class=\\"w-1/2 text-[14px] font-semibold pt-[15px] cursor-pointer\\"><div data-testid=\\"edit-car-details-link\\" class=\\"inline-block text-[#2E4341]\\">Edit car details<img class=\\"inline-block px-[4px] pb-[2px]\\" src=\\"edit.svg\\" alt=\\"edit icon\\"></div></div><div class=\\"w-1/2\\"><button class=\\"root StyledButton-sc-1apep4b-0 jVvup !text-[13px] !px-[17.25px] !normal-case whitespace-nowrap font-semibold MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"car.overview-button-confirm\\"><div style=\\"visibility: visible;\\">Confirm car details</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></div>"`;
